### ISSUE TITLE
GM Behavior when flag dropped.

### ISSUE THREAD

**Issue by bzWarPig** [BZFlag-Dev/bzflag#24]
When GM lock is established and shot fired, and the flag is dropped before the shot ends, either initiated by the player, or due to a flag shot limit, the GM lock is lost.  GM's should retain lock until either the shooter is killed or the shot ends.

**Comment by JMakey** [comment]
Pro tip: after the GM lock is lost due to this bug, you can right-click or press "i" to lock the shot again.

**Comment by JeffM2501** [comment]
I had a discussion about this with @blast007 ,trying to decide what is and isn't a bug, since there was no specification for how GMs should work.

The specification he provided is that the lock SHOULD be lost on drop or death, and the tank should NOT be able to re-lock if it does not have the flag. The reasoning is that the locking "equipment" is part of the flag, not part of the tank, and the shot will not get any more steering input if the tank does not have the flag.

I am in the process of moving GM shot logic and updates to the server, and will implement this specification. Since it will be server side, the behavior when the flag is lost will be changeable with server only changes or plugins so we can play-test different options to see what is the most fun.

I needed the specification because part of the changes I am making is allowing the server to create shots for a player as a proxy, so I needed to know how server created GMs should work.

**Comment by JeffM2501** [comment]
Depends on #120

**Comment by JeffM2501** [comment]
Fixed with commit 379bcfe34ae80f8fb5438139a7d0f70d60a9a829

GM can no longer be locked if the player does not have the GM flag, per specification.
